### ISSUE TITLE
Buttons: State visualisation and overlay toggling

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#21]
Most off the overlays listed in ```Mod settings > Overlays tab``` correspond to buttons on the in-game toolbar.

Would it be possible to right click or shift+click (or whatever) a button on the in-game toolbar to toggle its associated overlay?

Furthermore, is it possible to overlay a border image round the tools that have persistent overlays enabled?

Extending that, would it be worth changing background colour of active toolbar buttons?

Each button would have the following sprites:

* Inactive
* Active
* Inactive + Overlay
* Active + Overlay

Code changes required:

* Handing a click modifier (or right click) on the button to toggle persistent overlay
* Displaying/changing sprites on the in-game toolbar

I could do some quick image mockups this week to illustrate what it might look like?

**Comment by aubergine10** [comment]
It looks like the buttons on the toolbar also have a 'hover' state, am I correct?

Also, I read that there are plans to add some extra despawn tools to the toolbar - do you have a list of what those might be?

**Comment by krzychu124** [comment]
Yup, these buttons have hover state and they are created from atlas texture "Resources\mainmenu-btns.png"

Yes I've had a plan to add extra despawn buttons (Reset stuck cims and vehicles and Remove parked vehicles) because currently these tools are available only from TMPE Options screen. I have to do some tests to check if its possible.
Btw I have no idea how those icons should look like 😃

**Comment by aubergine10** [comment]
Ok, I'll see if I can find suitable icons tomorrow.

For the texture atlas, it looks like it's using 50x50px tiles, is that correct? And I assume the icons within those tiles are something like 48x48px?

I would like to attempt doing better icons for everything in that atlas, as currently some of them are a bit confusing.

With regards to the various despawning things (there will be 4 despawn-related buttons in total) I was wondering if it would be worth adding an option in settings to hide them all? I rarely if ever use them, so I wouldn't want them cluttering the toolbar.

Also, it might be worth providing a quick way to the mod settings from the main TMPE button (eg. right-click the crown button). The mod [total autobudget](https://steamcommunity.com/sharedfiles/filedetails/?id=1541897355&searchtext=) could be investigated as an example of how in-game UI (budget screen in its case) buttons could directly open the mod settings screen without having to go through "pause menu".

**Comment by krzychu124** [comment]
I will check that mod maybe there is something that could help.
With regards to the despawn-related buttons I've been thinking about something like we have one general button 'despawn' then after clicking we will see all 4 in row, column or in the other layout. This would be usable and compact. I will have to search is there any mod which has similar feature implemented but if not I will try to extend functionality of standard menu to be able display icons(buttons) instead of text. Quick example:
<IMG src="https://raw.githubusercontent.com/baronchen/ng2-float-btn/master/screenshot.gif" width=200>

**Comment by aubergine10** [comment]
I think some of the existing buttons could be merged too, For example, **Junction Restrictions**, **Priority Signs** and **Toggle Traffic Lights** could feasibly be grouped under a single button. While the three features would remain individually activatable in mod settings, the end-user visualisation could depict them as a single feature. So you go in to junction mode and in that one interface you get:

From 'junction restrictions':

* Enter blocked lanes
* U-turns
* Pedestrian crossings
* Straight-on lane changes

From 'priority signs':

* Stop / yield / priority

From 'toggle traffic lights':

* Toggle traffic lights

It would require better UI for all three features, but much of that could be achieved by looking at how the vanilla junction editor (stop signs and lights toggle) works and extending that UI to include the other features/icons.

**Comment by aubergine10** [comment]
If we go down the "build bar" route (see: #51), we could have dedicated tab to toggle overlays.
